vendor/bat-native-ads/src/bat/ads/internal/ad_events/ad_event_util.cc

```diff
@@ -23,4 +23,39 @@ bool HasFiredAdViewedEvent(const AdInfo& ad, const AdEventList& ad_events) {
   return true;
 }
 
+absl::optional<base::Time> GetLastSeenAdTime(
+    const AdEventList& ad_events,
+    const CreativeAdInfo& creative_ad) {
+  const auto iter = std::find_if(
+      ad_events.begin(), ad_events.end(),
+      [&creative_ad](const AdEventInfo& ad_event) -> bool {
+        return (ad_event.creative_instance_id ==
+                    creative_ad.creative_instance_id &&
+                ad_event.confirmation_type == ConfirmationType::kViewed);
+      });
+
+  if (iter == ad_events.end()) {
+    return absl::nullopt;
+  }
+
+  return base::Time::FromDoubleT(iter->timestamp);
+}
+
+absl::optional<base::Time> GetLastSeenAdvertiserTime(
+    const AdEventList& ad_events,
+    const CreativeAdInfo& creative_ad) {
+  const auto iter = std::find_if(
+      ad_events.begin(), ad_events.end(),
+      [&creative_ad](const AdEventInfo& ad_event) -> bool {
+        return (ad_event.advertiser_id == creative_ad.advertiser_id &&
+                ad_event.confirmation_type == ConfirmationType::kViewed);
+      });
+
+  if (iter == ad_events.end()) {
+    return absl::nullopt;
+  }
+
+  return base::Time::FromDoubleT(iter->timestamp);
```

Comment:
`timestamp` is an `int64_t` so we will need to change to `static_cast<double>(iter->timestamp)`